src/core/classes/drop-claim-conditions.ts

```diff
@@ -205,31 +209,37 @@ export class DropClaimConditions<
         this.storage,
       );
 
-      const contractType = ethers.utils.toUtf8String(
-        await this.contractWrapper.readContract.contractType(),
-      );
-
       try {
-        const [validMerkleProof] = this.isSignatureDrop(
-          this.contractWrapper.readContract,
-          contractType,
-        )
-          ? await this.contractWrapper.readContract.verifyClaimMerkleProof(
-              activeConditionIndex,
-              addressToCheck,
-              quantity,
-              {
-                proof: proofs.proof,
-                maxQuantityInAllowlist: proofs.maxClaimable,
-              },
-            )
-          : await this.contractWrapper.readContract.verifyClaimMerkleProof(
-              activeConditionIndex,
-              addressToCheck,
-              quantity,
-              proofs.proof,
-              proofs.maxClaimable,
-            );
+        let validMerkleProof;
+        if (
+          hasFunction<ContractMetadataContract>(
+            "contractType",
+            this.contractWrapper,
+          )
+        ) {
+          const wrapper = this.contractWrapper.readContract as
+            | DropERC721
+            | DropERC20;
+          [validMerkleProof] = await wrapper.verifyClaimMerkleProof(
+            activeConditionIndex,
+            addressToCheck,
+            quantity,
+            proofs.proof,
+            proofs.maxClaimable,
+          );
+        } else {
+          const wrapper = this.contractWrapper.readContract as BaseDropERC721;
```

Comment:
I think if you typeguard as `DropERC721 | DropERC20` above then this else will also be type guarded?

Reply:
Doesn't look like it type guards the else, guess it doesn't do set difference for the unioned types - it does work for the if part though at least